cmd/protoc-gen-crd/main.go

```diff
@@ -74,15 +75,30 @@ func generate(request *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorRespon
 
 	m := protomodel.NewModel(request, false)
 
-	filesToGen := make(map[*protomodel.FileDescriptor]bool)
+	legacyChannelFilesToGen := make(map[*protomodel.FileDescriptor]bool)
+	standardChannelFilesToGen := make(map[*protomodel.FileDescriptor]bool)
+	experimentalChannelFilesToGen := make(map[*protomodel.FileDescriptor]bool)
+
 	for _, fileName := range request.FileToGenerate {
 		fd := m.AllFilesByName[fileName]
 		if fd == nil {
 			return nil, fmt.Errorf("unable to find %s", request.FileToGenerate)
+		} else if strings.HasSuffix(fd.GetPackage(), "v1") {
+			standardChannelFilesToGen[fd] = true
+			log.Println("it is standard: ", fd)
+		} else if strings.HasSuffix(fd.GetPackage(), "v1alpha1") {
+			experimentalChannelFilesToGen[fd] = true
+			log.Println("it is experimental: ", fd)
 		}
-		filesToGen[fd] = true
+		// Legacy channel will have all files that are in standard and experimental
+		legacyChannelFilesToGen[fd] = true
```

Comment:
I go back and forth on whether or not we need a legacy channel. We can keep it for now and discuss later